### ISSUE TITLE
Get Audio bytes when `ReceiveUnmixedMeetingAudio` is set to `true`

### ISSUE THREAD

**Issue by muradhaider5786** [microsoftgraph/microsoft-graph-comms-samples#431]
**Describe the issue**
I am working on a bot that is supposed to attend group calls/online meetings. The bot needs to access audio spoken by each participant that will be send to transcription service at the end of the meeting. In `CallHandler.cs` we can add an event handler `OnAudioMediaReceived(object sender, AudioMediaReceivedEventArgs e)`, where we can get the audio buffers for each speaker. My question is, where can I find the byte array against each speaker. When event handler is fired `e.Buffer.Data` always returns '0's while there are audio bytes in ` e.Buffer.UnmixedAudioBuffers[0].Data`. So I am creating a byte array out of it and adding that to a generic list which will be parsed once the meeting is ended, to generate the transcript:
                 ```
  byte[] managedByteArray = new byte[e.Buffer.UnmixedAudioBuffers[0].Length];
                    int length = (int)e.Buffer.UnmixedAudioBuffers[0].Length;
                    Marshal.Copy(e.Buffer.UnmixedAudioBuffers[0].Data, managedByteArray, 0, length);
```
I just want to make sure that am I getting the audio bytes in a correct way? as I am not getting desired results from speech to text service ( I am using cognitive speech service provided by Microsoft for transcription) .

**Comment by 1fabi0** [comment]
Yes this is the correct way (from there you can pack the samples into wav files if you wish to listen to the audio) maybe you have to setup you cognitive service to inform it that you are sending pcm samples

**Comment by muradhaider5786** [comment]
I have tried to create a wav file out of it but no able to run it as it simply says:
![image](https://user-images.githubusercontent.com/79193192/111436998-6e0f4900-8724-11eb-9b3e-7b71de1fbd7a.png)
My code to create audio file simply accepts the list of byte arrays that I just showed in the question. Following is the code to generate the wave file out of it:
```
private string CreateAudioFile()
        {
            string startupPath = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().GetName().CodeBase);
            string rootPath = startupPath.Replace("file:\\", "");
            string filePath = rootPath.Replace("bin\\x64\\Debug", "speakerAudio" + fileNameCount + ".wav");
                return filePath;
        }

        private void SaveDataToAudioFile(string filePath, List<byte[]> data)
        {
            using (var stream = new FileStream(filePath, FileMode.Append))
            {
                foreach(byte[] byteArray in data)
                {
                    stream.Write(byteArray, 0, byteArray.Length);
                }
                
            }
        }
```

**Comment by 1fabi0** [comment]
You are missing the wav file header pleas checkout https://github.com/microsoft/BotBuilder-RealTimeMediaCalling/issues/19#issuecomment-311433357

**Comment by muradhaider5786** [comment]
Regarding, 

> maybe you have to setup you cognitive service to inform it that you are sending pcm samples
Where can we specify that we are sending PCM samples? as it might be expecting wav samples

**Comment by 1fabi0** [comment]
There aren't wav samples because wav is a file format as a container for pcm data but I don't know how to setup the cognitive service I think that's something on the cognitive service site

**Comment by muradhaider5786** [comment]
Hi @1fabi0,
Thanks for your help and responses, I am able to create an audible wav file out of received audio buffers. But facing an issue in a particular scenario while converting speech to text. When participant doesn't speak and is not muted as well, we are still getting audio buffers although those doesn't contain any voice/sound in it. Wav file is created but when this file is sent to Speech-to-text service, service times out as there seems to be a silence timeout limit for Microsoft speech to text service. So, is it possible to discard any silent buffers when participant is not mute?

**Comment by 1fabi0** [comment]
Audio Packages have a IsSilence flag so you should check this flag if it is set to false/to true. If you still receive audio packages(IsSilence is false) you may have to check for signatures on the receiving packages and filter these packages out but this can be difficult if the packages you receive are different

**Comment by stefan2410** [comment]
@1fabi0  The isSilence = true  flag  happens only initially  before the actual connection of the speakers.

**Comment by 1fabi0** [comment]
I had a similar problem with channel meetings but as I described check for patterns/signatures in the packages where the IsSilence flag is false ideally packages that are just silent have the flag is silence = true
